tests/integration/state_machine/test_state_machine.py

```diff
@@ -1,88 +1,66 @@
 import time
+from collections import deque
 from threading import Thread
 
+import pytest
+
 from isar.models.mission import Mission
 from isar.services.utilities.scheduling_utilities import SchedulingUtilities
 from isar.state_machine.state_machine import StateMachine, States, main
 from isar.storage.storage_interface import StorageInterface
 from robot_interface.models.mission import DriveToPose, MissionStatus, Step
 from robot_interface.robot_interface import RobotInterface
-from tests.mocks.blob_storage import StorageMock
 from tests.test_utilities.mock_interface.mock_robot_interface import MockRobot
 from tests.test_utilities.mock_models.mock_robot_variables import mock_pose
 from tests.test_utilities.mock_models.mock_step import MockStep
 
 
-def start_state_machine_in_thread(injector) -> StateMachine:
-    state_machine_thread: Thread = Thread(target=main, args=[injector])
-    state_machine_thread.daemon = True
-    state_machine_thread.start()
-    state_machine: StateMachine = injector.get(StateMachine)
-    return state_machine
+class StateMachineThread(object):
+    def __init__(self, injector) -> None:
+        self.injector = injector
+        injector.binder.bind(RobotInterface, to=MockRobot())
```

Comment:
With the new way of setting up injector modules the binder should no longer be necessary. However, it requires the implementation of the `MockRobotModule` similar to `MockStorageModule` [here](https://github.com/equinor/isar/blob/main/tests/test_modules.py).

Then you would set the injector to use `MockRobotModule` [here](https://github.com/equinor/isar/blob/affbb4ed3475c31b95b6c5be8fc6ad50d79471d0/tests/conftest.py#L44) instead of `RobotModule`.

Reply:
Since the `RobotModule` should be mocked in all tests it would be fine to do this in the base `injector`. For advanced tests it's possible to then setup a tailored `injector` which can pick and choose what should be mocked.

This can be a new issue if you wish or you can add it as part of this PR.

Reply:
Good catch, will fix!